easybuild/tools/module_generator.py

```diff
@@ -213,6 +214,79 @@ def swap_module(self, mod_name_out, mod_name_in, guarded=True):
         """
         raise NotImplementedError
 
+    def _generate_help_text(self):
+        """
+        Generate syntax-independent help text used for `module help`.
+        """
+        # General package description (mandatory)
+        description = self.app.cfg['description']
+        lines = [
+            '',
+            'Description',
+            '===========',
+            "%s" % description.strip(),
+        ]
+
+        # Package usage instructions (optional)
+        usage = self.app.cfg['usage']
+        if usage:
+            lines.extend([
+                '',
+                '',
+                'Usage',
+                '=====',
+                "%s" % usage.strip(),
+            ])
+
+        # Additional information
+        homepage = self.app.cfg['homepage']
+        lines.extend([
+            '',
+            '',
+            'More information',
+            '================',
+            " - Homepage: %s" % homepage,
+        ])
+
+        docpaths = self.app.cfg['docpaths'] or []
+        docurls = self.app.cfg['docurls'] or []
+        if docpaths or docurls:
+            env_name = convert_name(self.app.name, upper=True)
+            root_envvar = ROOT_ENV_VAR_NAME_PREFIX + env_name
+            lines.extend([" - Documentation:"])
+            lines.extend(["    - $%s/%s" % (root_envvar,path) for path in docpaths])
+            lines.extend(["    - %s" % url for url in docurls])
+
+        support = self.app.cfg['support']
+        if support:
+            if isinstance(support, list):
+                lines.extend([" - Support/bug reports:"])
+                lines.extend(["    - %s" % address for address in support])
+            else:
+                lines.extend([" - Support/bug reports: %s" % support])
+
+        contact = self.app.cfg['contact']
+        if contact:
+            if isinstance(contact, list):
+                lines.extend([" - Site contacts:"])
+                lines.extend(["    - %s" % address for address in contact])
+            else:
+                lines.extend([" - Site contact: %s" % contact])
+
+        # Extensions (if any)
+        exts_list = self.app.cfg['exts_list']
+        if exts_list:
+            extensions = ', '.join(sorted(['%s-%s' % (ext[0], ext[1]) for ext in exts_list], key=str.lower))
+            lines.extend([
+                '',
+                '',
+                "Included extensions",
+                '===================',
+                "%s" % '\n'.join(wrap(extensions, 78)),
```

Comment:
To make it fit nicely into a 80-character-wide terminal window :grin:

When making this a configuration parameter, I'd also like to see all other sections to be wrapped at the given width. But this is (at least from my "I don't know much Python" perspective) by no means trivial to achieve with `textwrap.wrap()`.  Maybe someone knows a good rst-to-ascii-text renderer that we could leverage?